### ISSUE TITLE
Support fail on warning (e.g. in pre-commit hook)

### ISSUE THREAD

**Issue by ccastillo232** [sasjs/lint#132]
Currently I have a pre-commit set up but when there are warnings it will commit the changes with no indication of any issues.  Only if I introduce a linting issue with severity of ERROR will the commit fail.

I want strict linting in my project and would like to add a pre-commit hook that fails on warnings from sasjs lint.  (I have ghooks installed - perhaps it's a ghooks setting that I am not seeing?)

As a workaround I would like to be able to configure the severity of linting issues to force some issues to be errors.

**Comment by allanbowe** [comment]
Update - we are currently working on this, and plan to implement a new object in the `.sasjslint` file that allows configuration of severity levels for each rule, eg:

```json
{
  "noTrailingSpaces": true,
  "hasDoxygenHeader": true,
  "maxLineLength": 300,
  "severityLevel": {
    "hasDoxygenHeader": "warn",
    "maxLineLength": "error",
    "noTrailingSpaces": "error"
  }
}
```

* "warn" - show warning in the log (doesn’t affect exit code)
* "error" - show error in the log (exit code is 1 when triggered)

If a rule is not specified, then the default severity level will be applied.

**Comment by allanbowe** [comment]
Hi @ccastillo232  - this issue was just addressed in this PR:  https://github.com/sasjs/lint/pull/181

It will be available in the next VS Code Extension release (it needs to be bumped to include this version of the lint)

**Comment by allanbowe** [comment]
CLI is working:

<img width="629" alt="image" src="https://user-images.githubusercontent.com/4420615/202473134-b930bbf3-94cc-4eb3-af5e-fe2c47b7c393.png">


and extension:

<img width="1336" alt="image" src="https://user-images.githubusercontent.com/4420615/202473320-dea8f64d-d74a-404c-8ef9-dbe75986b2e6.png">
